draft-ietf-oauth-selective-disclosure-jwt.md

```diff
@@ -224,11 +224,13 @@ Key Binding is an optional feature. When Key Binding is required by the use-case
 
 Note: How the public key is included in SD-JWT is out of scope of this document. It can be passed by value or by reference.
 
-The Holder can then create a signed document, the Key Binding JWT as defined in (#kb-jwt), using its private key. This document contains some
-data provided by the Verifier such as a nonce to ensure the freshness of the signature, and audience to indicate the
-intended audience for the document.
-
-The Key Binding JWT can be included as part of the SD-JWT and sent to the Verifier as described in (#data_formats).
+For presenting an SD-JWT to a Verifier that enforces Key Binding, the Holder
+creates a signed document, the Key Binding JWT as defined in (#kb-jwt), using
+its private key. This document contains some data provided by the Verifier, such
+as a nonce, to ensure the freshness of the signature, an audience value to
+indicate the intended audience for the document, and a hash that ensures the
+integrity of the data sent to the Verifier. Details of the format of Key Binding JWTs are
```

Comment:
The hash isn't provided by the Verifier so suggest breaking the hash part out of the sentence that starts "This document contains some data provided by the Verifier".  Also be a bit more specific that it's integrity of the data that the holder sent to the Verifier.  Something like: 

```suggestion
its private key. This document contains some data provided by the Verifier, such
as a nonce, to ensure the freshness of the signature, and an audience value to
indicate the intended audience for the document. It also includes a hash that ensures the
integrity of the data sent from the Holder to the Verifier. Details of the format of Key Binding JWTs are
```

Reply:
But the audience value is also not "provided" by the Verifier... so....
```suggestion
its private key. This document usually contains a nonce to ensure the freshness of the signature, an audience value to
indicate the intended audience for the document, and a hash that ensures the
integrity of the data sent to the Verifier. Details of the format of Key Binding JWTs are
```

Reply:
Yeah, the audience value isn't "provided" by the Verifier exactly but it could be from the verifier's metadata or HTTPS cert or something so does kinda come from the verifier. Also "provided" was used in the text prior to this PR so I was just keeping with that. But you are right (as you usually are!). Here's some proposed text riffing on your text that drops "usually" because the KB-JWT is defined to have these things and is a bit more specific that it's integrity of the data that the holder sent to the Verifier.

```suggestion
its private key. This document contains a nonce to ensure the freshness of the signature, an audience value to
indicate the intended audience for the document, and a hash that ensures the
integrity of the data sent from the Holder to the Verifier. Details of the format of Key Binding JWTs are
```